validations.py

```diff
@@ -0,0 +1,180 @@
+import argparse
+import glob
+import re
+import subprocess
+import yaml
+
+
+def get_fhir_resource_files(base_dir, recently_changed):
+    if recently_changed is True:
+        git_call = subprocess.run(['git', 'diff', '--name-only', 'master...'], check=True, capture_output=True)
+        grep_call = subprocess.run(['grep', base_dir], input=git_call.stdout, capture_output=True)
+        unique_call = subprocess.run(['uniq'], input=grep_call.stdout, capture_output=True)
+        output = unique_call.stdout.decode('utf-8')
+        if len(output) == 0:
+            return []
+        return output.strip().split('\n')
+    else:
+        return glob.glob(f'{base_dir}/*.json')
+
+
+def any_filters_match(patterns, error):
+    return any(re.match(regex, error) for regex in patterns)
+
+
+def get_global_filter(ignore_list):
+    global_filters = []
+
+    if type(ignore_list) is dict:
+        if 'global_filter' in ignore_list and type(ignore_list['global_filter']) is dict:
+            global_filter = ignore_list['global_filter']
+
+            if 'error_patterns' in global_filter and type(global_filter['error_patterns']) is list:
+                global_filters = global_filter['error_patterns']
+
+    return global_filters
+
+
+def get_file_filter(ignore_list, file_path):
+    filters = []
+
+    if type(ignore_list) is dict:
+        if 'file_filter' in ignore_list and ignore_list['file_filter'] is not None:
+            for file_filter in ignore_list['file_filter']:
+                if re.search(file_filter['file_pattern'], file_path):
+                    if 'error_patterns' in file_filter and type(file_filter['error_patterns']) is list:
+                        filters = filters + file_filter['error_patterns']
+
+    return filters
+
+
+def filter_errors(ignore_list, errors_per_file):
+    global_filters = get_global_filter(ignore_list)
+
+    filtered_errors_per_file = {}
+
+    for file_name in list(errors_per_file.keys()):
+        file_filters = get_file_filter(ignore_list, file_name)
+
+        for error in errors_per_file[file_name]:
+            if not any_filters_match(global_filters, error) and not any_filters_match(file_filters, error):
+                filtered_errors_per_file.setdefault(file_name, []).append(error)
+
+    return filtered_errors_per_file
+
+
+def validate_resources(version, ignore_list, files):
+    java_commands = ['java', '-Xmx3G', '-Xms2G', '-jar', 'validator_cli.jar']
+
+    for file_name in files:
+        java_commands.append(file_name)
+
+    java_commands.append('-version')
+    java_commands.append(version)
+
+    java_call = subprocess.run(java_commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output = java_call.stdout.decode('utf-8')
+    error_output = java_call.stderr.decode('utf-8')
+    if java_call.returncode != 0:
+        if output == '':
+            print('Validation of \'{}\' failed, but no output was generated.'.format(file_path))
```

Comment:
Why not exit in this case as well?  Maybe the indent level of the exit should be reduced so it always applies when `returncode` is not zero?

Reply:
Actually, I wasn't thinking when I looked at this before, the validator returns a non 0 exit code when it finds errors, which is what we're looking for.  These are different from runtime errors in the validator which is what is written to the error out stream, which is an all stop case.
Can probably still exit when the output is empty as well, as that's not supposed to happen.